new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1148 @@
+                                // location to write
+                                // domain
+                                .mountPath("/shared"))))) // mounted under /shared inside pod
+                    .volumes(Arrays.asList(
+                        new V1Volume()
+                            .name(pvName)
+                            .persistentVolumeClaim(
+                                new V1PersistentVolumeClaimVolumeSource()
+                                    .claimName(pvcName)),
+                        new V1Volume()
+                            .name("create-weblogic-domain-job-cm-volume")
+                            .configMap(
+                                new V1ConfigMapVolumeSource()
+                                    .name(domainScriptCM)))) //config map containing domain scripts
+                    .imagePullSecrets(isUseSecret ? Arrays.asList(
+                        new V1LocalObjectReference()
+                            .name(OCR_SECRET_NAME))
+                        : null))));
+    String jobName = assertDoesNotThrow(()
+        -> createNamespacedJob(jobBody), "Failed to create Job");
+
+    logger.info("Checking if the domain creation job {0} completed in namespace {1}",
+        jobName, namespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for job {0} to be completed in namespace {1} "
+                + "(elapsed time {2} ms, remaining time {3} ms)",
+                jobName,
+                namespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(jobCompleted(jobName, null, namespace));
+
+    // check job status and fail test if the job failed to create domain
+    V1Job job = assertDoesNotThrow(() -> getJob(jobName, namespace),
+        "Getting the job failed");
+    if (job != null) {
+      V1JobCondition jobCondition = job.getStatus().getConditions().stream().filter(
+          v1JobCondition -> "Failed".equalsIgnoreCase(v1JobCondition.getType()))
+          .findAny()
+          .orElse(null);
+      if (jobCondition != null) {
+        logger.severe("Job {0} failed to create domain", jobName);
+        List<V1Pod> pods = assertDoesNotThrow(()
+            -> listPods(namespace, "job-name=" + jobName).getItems(),
+            "Listing pods failed");
+        if (!pods.isEmpty()) {
+          String podLog = assertDoesNotThrow(() -> getPodLog(pods.get(0).getMetadata().getName(), namespace),
+              "Failed to get pod log");
+          logger.severe(podLog);
+          fail("Domain create job failed");
+        }
+      }
+    }
+
+  }
+
+  /**
+   * Create a persistent volume.
+   *
+   * @param pvName name of the persistent volume to create
+   * @param domainUid domain UID
+   * @throws IOException when creating pv path fails
+   */
+  private void createPV(String pvName, String domainUid) {
```

Comment:
Common Test Utility candidate

Reply:
Like Bhavani's comments I need to refactor some more classes to use methods from CommonTestUtils for the operations. Will do it in separate PR.